src/App.tsx

```diff
@@ -1,223 +1,172 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback, useContext, useEffect, useMemo, useState,
+} from 'react';
+import { deleteTodoById, getTodos, updateTodoById } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { CompletedFilter } from './types/CompletedFilter';
+import { Header } from './components/Header';
+import { Todolist } from './components/Todolist';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { filteredTodos } from './helper/Helper';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [completedFilter, setCompletedFilter] = useState(CompletedFilter.All);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [deletingTodosIds, setDeletingTodosIds] = useState([0]);
+  const [updatingTodosIds, setUpdatingTodosIds] = useState([0]);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const showError = useCallback((message: string) => {
+    setErrorMessage(message);
+
+    setTimeout(() => setErrorMessage(''), 3000);
+  }, []);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => showError('Can\'t load todos'));
     }
   }, []);
 
+  const closeErrorMessage = useCallback(() => {
+    setErrorMessage('');
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return filteredTodos(todos, completedFilter);
+  }, [todos, completedFilter]);
+
+  const updateTodo = (todoToChange: Todo) => {
```

Comment:
You are passing this callback to another component, so you should wrap it with useCallback

Reply:
done